airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/partition_enqueuer.py

```diff
@@ -50,6 +51,7 @@ def generate_partitions(self, stream: AbstractStream) -> None:
                 while self._thread_pool_manager.prune_to_validate_has_reached_futures_limit():
                     time.sleep(self._sleep_time_in_seconds)
                 self._queue.put(partition)
-            self._queue.put(PartitionGenerationCompletedSentinel(stream))
+                has_generated_partitions = True
+            self._queue.put(PartitionGenerationCompletedSentinel(stream, has_generated_partitions))
```

Comment:
Would it be more useful to have the partition count instead of a bool, down the line?

Reply:
I asked myself the same question and didn't see benefits today for that. There is a (very minor) drawback and it's using a int take more space and this space can grow as the int gets bigger

Reply:
Okay yeah not much of a drawback. Given that this feels like this is something we'll want when we add record counts and debug logging, I feel like it makes sense to go ahead and implement it as an int.

Reply:
See https://github.com/airbytehq/airbyte/pull/34605#issuecomment-1915039891 which seems somewhat relevant to this comment

Reply:
Right - I read the drawbacks that you listed for that approach and agree that the sentinel approach seems preferable.